src/shell/dotnet/Shell/App.xaml.cs

```diff
@@ -175,4 +178,24 @@ private void InjectMessageRouterConfig()
                     
                 """);
     }
+   
+    private void InjectFDC3()
+    {
+        string bundle = @"fdc3-iife-bundle.js";
+        string iife = ReadResource(bundle);
+
+        WebWindow.AddPreloadScript(iife);
+    }
+
+    public string ReadResource(string name)
+    {
+        var assembly = Assembly.GetExecutingAssembly();
+        string resourcePath = $$"""Shell.Preload.{{name}}""";
+        
+        using (Stream stream = assembly.GetManifestResourceStream(resourcePath))
+        using (StreamReader reader = new StreamReader(stream))
+        {
+            return reader.ReadToEnd();
```

Comment:
This is picked up by the coverage report that it's not covered, and it sounds like a pretty relevant test for me - if the resource is renamed, or otherwise not loadable, it should fail in CI